src/hotspot/share/runtime/coroutine.hpp

```diff
@@ -491,6 +491,40 @@ class EnableStealMark: public StackObj {
   ~EnableStealMark();
 };
 
+// Spin lock for cocurrent coroutine support.
+// It is useless at safepoint.
+class CoroutineSupportLocker : public StackObj {
```

Comment:
Why do we need a spin lock? Isn't `Monitor` suitable?

Reply:
`_coroutine_list` may be traversed not in safepoint. If this list changed by steal at this time, that will lead to deadlock. So we need a spin lock to resolve this problem. This locker use `cas` in previous version. So we create a spin lock to follow it.